image_metadata_script.py

```diff
@@ -0,0 +1,53 @@
+import os, time
+import filetype
+import csv
+import sys
+
+
+def image_metadata(path):
+
+    minio_link = 'None'
+    dataset = 'None'
+    is_processed = 0
+
+    try:
+        output_file = open('file_name.csv', 'w')
+
+        fields = ['IV_Name', 'Camera_ID', 'IV_date', 'IV_time', 'File_type', 'File_size', 'Minio_link', 'Dataset',
+                  'is_processed']
+        writer = csv.writer(output_file)
+        writer.writerow(fields)
+
+        try:
+            dirs = os.listdir(path)
+        except Exception as e:
+            print("Path is not a valid directory")
+            sys.exit(1)
+
+        for subdir in dirs:
+            path2 = os.path.join(path, subdir)
+            if os.path.isfile(path2):
+                continue
```

Comment:
if you want your code to work in reality, for each entity in os.listdir(path)
If it is an image file, get and write its metadata, with cameraID empty or None; 
if it is a directory, say subdir_1, then go down one directory down and get all image files in subdir_1, write their metadata, with cameraID either in name of the folder or the name of the image file.

Reply:
Okay, I have made this change. The csv file now shows the metadata of an image file found inside os.listdir(path) with camera id set to 'None'.